### PR TITLE
docs: document using custom HTTP clients

### PR DESCRIPTION
**Pull request by emmenko**
Ref: #2553 
Parent PR: #2734

### CODE REVIEW

**Comment by CarlosCortizasCT** [comment]
It seems we would need to adjust some links because of changes in the "Data Fetching" document.
Perhaps update the ones which causes [the error in CI](https://github.com/commercetools/merchant-center-application-kit/runs/7631862415?check_suite_focus=true):
```
// BEFORE
/custom-applications/development/data-fetching#requests-to-rest-apis

// AFTER
/custom-applications/development/data-fetching#rest-http-client
```

File: website/src/content/concepts/integrate-with-your-own-api.mdx [hunk 1]
```diff
@@ -388,25 +403,20 @@ Versions follow an incremental number in the format `v1`, `v2`, etc.
 
 The current version refers to the **default** version used by the API, if no `Accept-version` header is provided.
 
-When we introduce a new version, the default version enters a **deprecation period**, after which the version is marked as deprecated and the new version becomes the **default** version.
-
-| Feature maturity     | Deprecation period |
-| -------------------- | ------------------ |
-| Beta                 | 4 months           |
-| General Availability | 12 months          |
```
Comment (emmenko) [inline website/src/content/concepts/integrate-with-your-own-api.mdx:396]: Unrelated, but we don't need this anymore

File: website/src/content/development/data-fetching.mdx [hunk 2]
```diff
@@ -128,6 +134,308 @@ const getErrorMessage = (error) =>
...
+
+The available proxy targets are exposed as a `MC_API_PROXY_TARGETS` constant in the [`@commercetools-frontend/constants`](/api-reference/commercetools-frontend-constants) package.
+
+<Warning>
+
+The SDK library does not include features like data normalization, caching, etc. You will need to build these on your own.
+
+</Warning>
+
+# Custom HTTP client
```
Comment (emmenko) [inline website/src/content/development/data-fetching.mdx:186]: This is the new section

File: website/src/content/concepts/merchant-center-api.mdx [hunk 3]
```diff
@@ -91,6 +101,26 @@ When you develop a Custom Application all authentication logic is handled implic
 
 </Info>
 
+# HTTP headers
+
+Accessing the Merchant Center API requires to include certain HTTP headers in the HTTP request.
+
+- `Accept`: Required.<br/>
```
Comment (CarlosCortizasCT) [inline website/src/content/concepts/merchant-center-api.mdx:108]: nitpick: Just a personal suggestion to use another format where description of the header is in the same line:
image

File: website/src/content/concepts/integrate-with-your-own-api.mdx [hunk 4]
```diff
@@ -24,12 +24,19 @@ To be able to validate that the request comes from an authenticated user, the Me
...
 
-- `Accept-version: v2`: See [Versioning](#versioning).
-- `X-Forward-To: <url>`: The URL to your external API.
-- `X-Forward-To-Audience-Policy: <policy>`: See [Configuring the audience policy](#configuring-the-audience-policy).
-- `X-Project-Key: <project-key>`: The key of the commercetools Project currently being used by the Custom Application. The Merchant Center API Gateway will perform a validation check to ensure that the user has access to the Project, then forward the request to your server only if the check was successful.
+- `Accept-version`: Required.<br/>
+  See [Versioning](#versioning).
+- `X-Forward-To`: Required.<br/>
+  Set it to the external API URL.
+- `X-Forward-To-Audience-Policy`: Required<br/>.
```
Comment (CarlosCortizasCT) [inline website/src/content/concepts/integrate-with-your-own-api.mdx:35]:
```suggestion
- `X-Forward-To-Audience-Policy`: Required.<br/>
```

**Review by CarlosCortizasCT — COMMENTED** [review]
Great new additions to the docs 👏 .

I just left some minor suggestions.

**Comment by emmenko** [comment]
@CarlosCortizasCT Thanks for the suggestions. Updated here: https://github.com/commercetools/merchant-center-application-kit/pull/2738/commits/f275961ac8c9807a14cce4fb69f0e39e757045d0

**Review by CarlosCortizasCT — APPROVED** [review]
💯

**Review by Rhotimee — APPROVED** [review]

**Review by ddouglasz — APPROVED** [review]
Very educating to follow through 👏

**Pull request merged**